packages/graphql-client/README.md

```diff
@@ -0,0 +1,153 @@
+# GraphQL Client
+
+The GraphQL Client is a generic GraphQL client that can be used directly to interact with a GraphQL API. Client users are expected to provide the full API URL and necessary headers.
+
+## Initialization
+
+```typescript
+import {createGraphQLClient} from '@shopify/graphql-client';
+
+const client = createGraphQLClient({
+  url: 'http://your-shop-name.myshopify.com/api/2023-10/graphql.json',
+  headers: {
+    'Content-Type': 'application/json',
+    'X-Shopify-Storefront-Access-Token': 'public-token',
+  },
+});
+```
+
+### `createGraphQLClient()` parameters
+
+| Property | Type                     | Description                        |
+| -------- | ------------------------ | ---------------------------------- |
+| url      | `string`                 | The Storefront API URL             |
+| headers  | `Record<string, string>` | Headers to be included in requests |
+
+## Client properties
+
+| Property      | Type                                                                                                                                                 | Description                                                                                                                                                                                                                                                                                                                                                                |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| config        | `ClientConfig`                                                                                                                                | Configuration for the client                                                                                                                                                                                                                                                                                                                                               |
```

Comment:
Where is `ClientConfig` used?

Reply:
It's not used by the code but it's meant to make initial client configs available to the client user - it's not a necessity but more of a nice to have.

Reply:
It is defined in the `types.ts` but I didn't want to link directly to the file. I can also change this documentation to just show the definition directly in here or add a new section for this interface - how do we do this in the shopify-api library? Is there a preference?

Reply:
I agree that linking to the code wouldn't be ideal. Could we just list out the fields here like we do for the other types?

Reply:
Yup - can do!